### DISCUSSION TITLE
Are there any APIs to fetch the subscription details and billing history and their cost and limits?

### DISCUSSION THREAD

**Discussion by khirthi-ncompass** [community/community#33792] [Enterprise]
The API must fetch the subscription details information such as the current user's plan, renewal type (Monthly/Annual), Renewal date, Billing card details, Billing emails etc.

The API must fetch the history of items by month/range along with their pdfs.

**Comment by nicl** [comment]
To add a possibly related question/use case. I'm currently looking for an API to get historical billing for GHA so that I can graph costs over time. All I can see at the moment is https://docs.github.com/en/rest/billing#get-github-actions-billing-for-an-organization which is only for the current billing period.

**Reply by byrneh** [reply]
There are some billing license APIs available
[https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/billing](https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/billing)

If you have and Enterpise Cloud account you shoud be able to access this API for licene details
[https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/license](https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/license)

nothing for history or generating pdfs that I am aware

**Reply by nicl** [reply]
Indeed, looks like it is unsupported alas.
